File: sandbox/test_project/backend/app/repositories/todo_repository.py
from app.schemas.todo import TodoRead


class TodoRepository:
    """Persistence boundary for todo data.

    The starter implementation remains in memory. Database overlays replace or extend this class.
    """

    def __init__(self) -> None:
        self._todos: dict[int, TodoRead] = {}
        self._next_id = 1

    def list(self) -> list[TodoRead]:
        return list(self._todos.values())

    def create(self, title: str) -> TodoRead:
        todo = TodoRead(id=self._next_id, title=title, completed=False)
        self._todos[todo.id] = todo
        self._next_id += 1
        return todo

File: sandbox/test_project_02/backend/src/main/java/dev/reversevibe/backend/TodoRepository.java
package dev.reversevibe.backend;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

class TodoRepository {
  private final Map<Integer, Todo> todos = new LinkedHashMap<>();
  private int nextId = 1;

  List<Todo> list() { return new ArrayList<>(todos.values()); }

  Todo create(String title) {
    Todo todo = new Todo(nextId, title, false);
    todos.put(nextId, todo);
    nextId += 1;
    return todo;
  }
}
